hwmv2:  arm64: intel_socfpga: Port to HWMv2

golowanow opened the pull request:
Move and convert to HWMv2 `soc/arm64/intel_socfpga` SoC family `INTEL_SOCFPGA`  with its SoC series  `AGILEX`, `AGILEX5` and related board configurations:
* `intel_socfpga_agilex_socdk`
* `intel_socfpga_agilex5_socdk`

golowanow committed:
soc: arm64: intel_socfpga: Move and convert to HWMv2

Move and convert soc/arm64/intel_socfpga SoC family
`intel_socfpga` configuration to HWMv2 with its SoC
series: `agilex` and `agilex5` and related SoCs:
`intel_socfpga_agilex` and `intel_socfpga_agilex5`.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: arm64: intel_socfpga_agilex_socdk: move to HWMv2

Move and convert to HWMv2 `intel_socfpga_agilex_socdk`
board configuration.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: arm64: intel_socfpga_agilex5_socdk: move to HWMv2

Move and convert to HWMv2 `intel_socfpga_agilex5_socdk`
board configuration.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented:
@gdengi, @nbalabak, @teikheng - up to your attention (for some reason I can't assign you as reviewers to this PR)

https://github.com/zephyrproject-rtos/zephyr/blob/8243429d7a4a5cf8b5613d694a2ad2bf57e2d62d/MAINTAINERS.yml#L3233-L3247

gdengi commented:
> @gdengi, @nbalabak, @teikheng - up to your attention (for some reason I can't assign you as reviewers to this PR)
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/8243429d7a4a5cf8b5613d694a2ad2bf57e2d62d/MAINTAINERS.yml#L3233-L3247

@golowanow sure - thank you. Let me go through these changes and update you.

nordicjm commented on soc/intel/intel_socfpga/agilex/Kconfig.soc:
```diff
@@ -0,0 +1,18 @@
+	  "Intel SoC FPGA Agilex"
+
+config SOC
+	default "intel_socfpga_agilex" if SOC_AGILEX
```
```suggestion
	default "agilex" if SOC_AGILEX
```

golowanow replied:
why a short name, just `agilex` ?

nordicjm replied:
Because SOC_AGILEX -> agilex

golowanow replied:
I assume this note is also related to SoC/board name optimization as this discussion - https://github.com/zephyrproject-rtos/zephyr/pull/68915#discussion_r1487770683

In regard of this PR my opinion is the same: it is better to avoid ad-hoc name changes doing such a bulk migration unless there is naming convention for all the boards and SoC of a vendor aligned to the HWMv2 scheme.

For these SoC/boards It is possible to keep just the same board names as currently.
I would left to the board maintainers to decide when (now?/later?) and how exactly to change these names if it makes sense.

nordicjm commented on soc/intel/intel_socfpga/agilex5/Kconfig.soc:
```diff
@@ -0,0 +1,18 @@
+	  "Intel SoC FPGA Agilex5"
+
+config SOC
+	default "intel_socfpga_agilex5" if SOC_AGILEX5
```
as above

nordicjm commented on soc/intel/intel_socfpga/soc.yml:
```diff
@@ -0,0 +1,9 @@
+    series:
+      - name: agilex
+        socs:
+          - name: intel_socfpga_agilex
```
as above

nordicjm commented on boards/intel/intel_socfpga_agilex5_socdk/Kconfig.intel_socfpga_agilex5_socdk:
```diff
@@ -0,0 +1,8 @@
+	bool
+	select SOC_AGILEX5
+	help
+	  "Intel SoC FPGA Development Kit (Agilex5)"
```
as above

golowanow replied:
changed

nordicjm commented on boards/intel/intel_socfpga_agilex5_socdk/Kconfig.intel_socfpga_agilex5_socdk:
```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_INTEL_SOCFPGA_AGILEX5_SOCDK
+	bool
```
```suggestion
```

golowanow replied:
removed

nordicjm commented on boards/intel/intel_socfpga_agilex5_socdk/board.yml:
```diff
@@ -0,0 +1,4 @@
+board:
+  name: intel_socfpga_agilex5_socdk
+  socs:
+    - name: intel_socfpga_agilex5
```
as above

golowanow commented on soc/intel/intel_socfpga/agilex/Kconfig:
```diff
@@ -1,10 +1,10 @@
-	bool "Intel SoC FPGA Agilex Series"
 	select ARM64
 	select CPU_CORTEX_A53
 	select SOC_FAMILY_INTEL_SOCFPGA
```
removed

golowanow commented on soc/intel/intel_socfpga/agilex5/Kconfig:
```diff
@@ -1,10 +1,10 @@
 	select SOC_FAMILY_INTEL_SOCFPGA
-	help
-		Enable support for Intel SoC FPGA Series
+
```
changed

golowanow commented on boards/intel/intel_socfpga_agilex_socdk/Kconfig.intel_socfpga_agilex_socdk:
```diff
@@ -0,0 +1,8 @@
+	bool
+	select SOC_AGILEX
+	help
+	  "Intel SoC FPGA Development Kit (Agilex)"
```
changed

nashif commented:
we are adding some duplication in the naming and constant repition of names, so something like this:

`intel_socfpga_agilex5_socdk` ends up using the long form being` intel_socfpga_agilex5_socdk/intel_socfpga_agilex5`.

We should look into introducing `intel_socfpga` as the board and `agilex5_socdk` and `agilex_socdk` as the soc, so we end up with names likes:

- intel_socfpga/agilex5_socdk
- intel_socfpga/agilex_socdk

gmarull reviewed (dismissed):
update docs/maintainers as needed

gdengi commented:
@golowanow  Shall we even keep the board directory structure as same as soc (soc/intel/intel_socfpga/agilex or agilex5) like 

_boards/intel/intel_socfpga/agilex_socdk 
boards/intel/intel_socfpga/agilex5_socdk_

instead of this
_boards /intel/intel_socfpga_agilex_socdk  
boards /intel/intel_socfpga_agilex5_socdk_

golowanow committed:
boards: arm64: intel_socfpga: Move boards to subdirectories

Adjust boards directory structure to SoC structure:

 `boards/intel/intel_socfpga_agilex_socdk` ->
     `boards/intel/intel_socfpga/agilex_socdk`
 `boards/intel/intel_socfpga_agilex5_socdk` ->
     `boards/intel/intel_socfpga/agilex5_socdk`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
MAINTAINERS: intel_socfpga: Adjust to HWMv2 move

Adjust intel/intel_socfpga to HWMv2 move.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented:
> We should look into introducing `intel_socfpga` as the board and `agilex5_socdk` and `agilex_socdk` as the soc, so we end up with names likes:
> 
> * intel_socfpga/agilex5_socdk
> * intel_socfpga/agilex_socdk

@gdengi what do you think to rename SoC/boards in the way as above ? now or later ?

golowanow committed:
soc: arm64: intel_socfpga: hwmv2: Rename SoCs

Rename SoC for better HWMv2 use:
  `intel_socfpga_agilex`  --> `agilex`
  `intel_socfpga_agilex5` --> `agilex5`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented:
> @gdengi what do you think to rename SoC/boards in the way as above ? now or later ?
> 

@gdengi, @nordicjm, @nashif  - I've added one commit to simplify SoC names:
`intel_socfpga_agilex` --> `agilex`
`intel_socfpga_agilex5` --> `agilex5`

Board names refactoring might be done later, if needed.
